index.d.ts

```diff
@@ -1,184 +1,261 @@
+  export interface UseSortbyOptions {
+    sortByFn?: (a: any, b: any, desc: boolean) => 0 | 1 | -1
+    manualSorting?: boolean
+    disableSorting?: boolean
+    defaultSortDesc?: boolean
+    disableMultiSort?: boolean
+  }
 
-    export function useColumns<D>(props: TableOptions<D>): TableOptions<D> & UseColumnsValues<D>;
+  export interface UseSortbyState<D> {
+    sortBy?: SortingRules<D>
+  }
 
-    export function useRows<D>(props: TableOptions<D>): TableOptions<D> & UseRowsValues<D>;
+  export interface TableInstance<D = any> extends TableOptions<D> {
+    hooks: Hooks
+    rows: Row<D>[]
+    columns: EnhancedColumn<D>[]
+    headerGroups: HeaderGroup<D>[]
+    headers: HeaderGroup<D>[]
+    getTableProps: (userProps?: any) => any
+    getRowProps: (userProps?: any) => any
+    prepareRow: (row: Row<D>) => any
+  }
 
-    export function useFilters<D>(
-        props: TableOptions<D>,
-    ): TableOptions<D> & {
-        rows: Row<D>[];
-    };
+  export interface TableOptions<D = any> {
+    data: D[]
+    columns: HeaderColumn<D>[]
+    state: State<D>
+    debug?: boolean
+    loading: boolean
+    defaultColumn?: Partial<Column<D>>
+  }
 
-    export function useSortBy<D>(
-        props: TableOptions<D>,
-    ): TableOptions<D> & {
-        rows: Row<D>[];
-    };
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  export interface TableState<D = any> {}
 
-    export function useGroupBy<D>(props: TableOptions<D>): TableOptions<D> & { rows: Row<D>[] };
+  export type SetState<D> = (
+    updater: (old: TableState<D>) => TableState<D>,
+    actions: any
+  ) => void
 
-    export function usePagination<D>(props: TableOptions<D>): UsePaginationValues<D>;
+  export type State<D> = [TableState<D>, SetState<D>]
 
-    export function useFlexLayout<D>(props: TableOptions<D>): TableOptions<D>;
+  export function useTable<D = any>(
+    props: TableOptions<D>,
+    ...plugins: any[]
+  ): TableInstance<D>
+
+  export function useFilters<D = any>(
+    props: TableOptions<D>
+  ): TableOptions<D> & {
+    rows: Row<D>[]
+  }
+
+  export function useSortBy<D = any>(
+    props: TableOptions<D>
+  ): TableOptions<D> & {
+    rows: Row<D>[]
+  }
+
+  export function useGroupBy<D = any>(
+    props: TableOptions<D>
+  ): TableOptions<D> & { rows: Row<D>[] }
+
+  export function usePagination<D = any>(
+    props: TableOptions<D>
+  ): UsePaginationValues<D>
+
+  export function useExpanded<D = any>(
+    props: TableOptions<D>
+  ): TableOptions<D> & {
+    toggleExpandedByPath: () => any
+    expandedDepth: []
+    rows: Row<D>[]
+  }
+
+  export function useTableState<D = any>(
+    initialState?: Partial<TableState<D>>,
+    overriddenState?:  Partial<TableState<D>>,
+    options?: {
+      reducer?: (
+        oldState: TableState<D>,
+        newState: TableState<D>,
+        type: string
+      ) => any
+      useState?: Dispatch<SetStateAction<any>>
```

Comment:
What do you think of using `ReturnValue<typeof useState<any>>`? I _think_ that's right? If not, this should be an array. AFAIK, this is the result of the `useState` hook?

Reply:
actually I think that it should just be `typeof useState` since it's a function and defaults to React.useState.

Reply:
You're correct. `typeof useState` should work